src/image.cpp

```diff
@@ -109,15 +158,7 @@ void image::convert_mix_channels(vector<cv::Mat>& source,
         prepared_source = &tmp_source;
     }
 
-    if (prepared_source->size() == 1 && target.size() == 1)
-    {
-        size_t size = target[0].total() * target[0].elemSize();
-        memcpy(target[0].data, (*prepared_source)[0].data, size);
-    }
-    else
```

Comment:
why did you remove that optimization?

Reply:
I don't remember. I think I could bring it back if I add a check that from_to is  something in the pattern of 0,0 ;    0,0, 1,1 ;     0,0, 1,1, 2,2;     0,0, 1,1, 2,2, 3,3. 
I don't know if it will work in this case https://github.com/NervanaSystems/aeon/pull/6/files#diff-a4ccfbf4e57f797ce3bbb8864ceaf6cbR310 where the target image is a cropped with cv::Rect matrix. I think the data is not contiguous there?

Reply:
I could actually check sth like `if (mat.step[0] == width*sizeof(data type)&& mat.step[1] == height*sizeof(data type))` to handle that case, right?

Reply:
I check by isContinuous